### Урок 8. Дисперсионный анализ. Логистическая регрессия

In [2]:
import numpy as np

1. Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [20]:
football = [173, 175, 180, 178, 177, 185, 183, 182]
hokkey = [177, 179, 180, 188, 177, 172, 171, 184, 180]
weightlifters = [172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170]
k = 3
football_n = len(football)
hokkey_n = len(hokkey)
weightlifters_n = len(weightlifters)
total_n = football_n + hokkey_n + weightlifters_n

In [21]:
football_av = sum(football)/football_n
hokkey_av = sum(hokkey)/hokkey_n
weightlifters_av = sum(weightlifters)/weightlifters_n
total_av = (sum(football) + sum(hokkey) + sum(weightlifters)) / total_n

In [37]:
total_av

176.46428571428572

In [39]:
S_2 = 0
for i in football:
    S_2 += (i - total_av) ** 2
for i in hokkey:
    S_2 += (i - total_av) ** 2
for i in weightlifters:
    S_2 += (i - total_av) ** 2
S_2

830.9642857142854

In [23]:
S_F = ((football_av - total_av) ** 2) * football_n + ((hokkey_av - total_av) ** 2) * hokkey_n + ((weightlifters_av - total_av) ** 2) * weightlifters_n
S_F

253.9074675324678

In [24]:
S_ost = 0
for i in football:
    S_ost += (i - football_av) ** 2
for i in hokkey:
    S_ost += (i - hokkey_av) ** 2
for i in weightlifters:
    S_ost += (i - weightlifters_av) ** 2
S_ost

577.0568181818181

In [25]:
S_F+S_ost

830.9642857142859

In [40]:
disp_f = S_F / (k - 1)
disp_ost = S_ost / (total_n - k)
K_fisher = disp_f / disp_ost
K_fisher

5.500053450812599

In [41]:
df1 = k - 1
df2 = total_n - k
print(df1, df2)

2 25


In [42]:
K_fisher_krit = 3.38 # при a = 0.05

5.50 > 3.38. Различия среднего роста среди взрослых футболистов, хоккеистов и штангистов есть при уровне значимости a = 0.05

In [43]:
from scipy import stats

In [44]:
y1 = np.array(football)
y2 = np.array(hokkey)
y3 = np.array(weightlifters)
stats.f_oneway(y1, y2, y3)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)